From b4bbeb2bcc4d35bc76a63c2ec00e4a1461214554 Mon Sep 17 00:00:00 2001
From: Soren Brinkmann <soren.brinkmann@xilinx.com>
Date: Mon, 12 Nov 2012 14:36:06 -0800
Subject: [PATCH 054/793] net: xilinx_emacps: Don't disable clocks if
 suspended

When entering suspend the device clocks were disabled, even
when they were already gated off due to RUNTIME_PM. Triggering
a kernel warning.

Signed-off-by: Soren Brinkmann <soren.brinkmann@xilinx.com>
Acked-by: Michal Simek <michal.simek@xilinx.com>
---
 drivers/net/ethernet/xilinx/xilinx_emacps.c |   12 ++++++++----
 1 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_emacps.c b/drivers/net/ethernet/xilinx/xilinx_emacps.c
index 39b83ff..14a3890 100644
--- a/drivers/net/ethernet/xilinx/xilinx_emacps.c
+++ b/drivers/net/ethernet/xilinx/xilinx_emacps.c
@@ -3316,8 +3316,10 @@ static int xemacps_suspend(struct device *device)
 #endif
 	netif_device_detach(ndev);
 #ifdef CONFIG_COMMON_CLK
-	clk_disable(lp->devclk);
-	clk_disable(lp->aperclk);
+	if (!pm_runtime_suspended(device)) {
+		clk_disable(lp->devclk);
+		clk_disable(lp->aperclk);
+	}
 #endif
 	return 0;
 }
@@ -3336,8 +3338,10 @@ static int xemacps_resume(struct device *device)
 #ifdef CONFIG_COMMON_CLK
 	struct net_local *lp = netdev_priv(ndev);
 
-	clk_enable(lp->aperclk);
-	clk_enable(lp->devclk);
+	if (!pm_runtime_suspended(device)) {
+		clk_enable(lp->aperclk);
+		clk_enable(lp->devclk);
+	}
 #endif
 	netif_device_attach(ndev);
 	return 0;
-- 
1.7.5.4

